src/main/java/uk/gov/justice/digital/delius/controller/StaffController.java

```diff
@@ -0,0 +1,71 @@
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import uk.gov.justice.digital.delius.data.api.AccessLimitation;
+import uk.gov.justice.digital.delius.data.api.ManagedOffender;
+import uk.gov.justice.digital.delius.jwt.Jwt;
+import uk.gov.justice.digital.delius.jwt.JwtValidation;
+import uk.gov.justice.digital.delius.service.NoSuchUserException;
+import uk.gov.justice.digital.delius.service.StaffService;
+
+import java.util.List;
+
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.OK;
+
+@RestController
+@Slf4j
+@Api(tags = "Staff")
+public class StaffController
+{
+    private StaffService staffService;
+    private Jwt jwt;
+
+    @Autowired
+    public StaffController(StaffService staffService, Jwt jwt) {
+        this.staffService = staffService;
+        this.jwt = jwt;
+    }
+
+    @RequestMapping(value = "/staff/staffCode/{staffCode}/managedOffenders", method = RequestMethod.GET)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "A list of managed offenders for a staff officer", response = ManagedOffender.class, responseContainer = "List"),
+            @ApiResponse(code = 401, message = "Request is missing Authorization header (no JWT)"),
+            @ApiResponse(code = 403, message = "The requesting user was restricted from access", response = AccessLimitation.class),
+            @ApiResponse(code = 404, message = "The requested staffCode was not found")
+    })
+    @JwtValidation
+    public ResponseEntity<List<ManagedOffender>> getManagedOffendersByStaffCode(final @RequestHeader HttpHeaders httpHeaders,
+                                                                                final @PathVariable("staffCode") String staffCode,
+                                                                                final @RequestParam(name="current", required=false, defaultValue="false") String current) {
+        return staffService.getManagedOffendersByStaffCode(staffCode, current)
+                .map(managedOffenders -> new ResponseEntity<>(managedOffenders ,OK))
+                .orElse(new ResponseEntity<>(NOT_FOUND));
+
+        // TODO: Check this - only NOT_FOUND if the staffCode was not an existing officer
+    }
+
+    @ExceptionHandler(HttpClientErrorException.class)
+    public ResponseEntity<String> restClientError(HttpClientErrorException e) {
+        return new ResponseEntity<>(e.getMessage(), e.getStatusCode());
+    }
+
+    @ExceptionHandler(HttpServerErrorException.class)
+    public ResponseEntity<String> restServerError(HttpServerErrorException e) {
+        return new ResponseEntity<>(e.getMessage(), e.getStatusCode());
+    }
+
+    @ExceptionHandler(NoSuchUserException.class)
+    public ResponseEntity<String> noSuchUser(NoSuchUserException e) {
+        return new ResponseEntity<>(e.getMessage(), NOT_FOUND);
+    }
```

Comment:
May want to consider moving these exception handlers into a new ControllerAdvice, alongside those in the OffenderController? Now that we have a new controller context?

Reply:
The only @ControllerAdvice I've seen in the application is for Jwt validation. For offenders, the exception handlers are as they are in the StaffController with local @ExceptionHandler methods, but yes, creating shared advice would save the controllers from duplicating these handlers so look at moving these.